client/src/components/Common/Header/Header.js

```diff
@@ -1,19 +1,48 @@
-import React from 'react'
+import React, { Component } from 'react'
 import GSGlog from '../../../Images/gsg_logo.jpg'
 import Profile from '../../../Images/Ellipse 2.jpg'
 import './Header.css'
 import { Link } from 'react-router-dom'
+// through withRouter You can get access to the history object’s properties 
+// withRouter will pass updated match,location, and history props to the wrapped component whenever it renders.
+import { withRouter } from 'react-router'
+import axios from 'axios'
 
-function Header () {
-  return (
-    <div className='header'>
-      <Link to={'/'} className='header_link'>
-        <img className='logo' src={GSGlog} alt='logo' />
-      </Link>
-      <Link to = {'/home'}>
-        <img className='profile' src={Profile} alt='profile'/></Link>
-    </div>
-  )
+class Header extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      email: '',
+      userId: '',
+      userName: ''
+    }
+  }
+
+  show = ()=> {
```

Comment:
not a very good name as it doesn't say what it's supposed to show, and actually does not show anything - `redirectToHome` or sth similar would be better.